### PR TITLE
Feature/routers Kacper

### PR DESCRIPTION
**Pull request by dubielel**

### CODE REVIEW

**Comment by kacienk** [comment]
Merge when the rest of endpoints are done

File: backend/models/__init__.py [hunk 1]
```diff
@@ -3,4 +3,4 @@
 from models.user import User  # noqa: F401
 from models.prescription import Prescription  # noqa: F401
 from models.referral import Referral  # noqa: F401
-from models.medical_documentation import MedicalDocumentation  # noqa: F401
+from models.medical_documentation import MedicalDocumentation, MedicalDocumentationEntry  # noqa: F401
```
Comment (kacienk) [inline backend/models/__init__.py:6]: Why did you add noqa F401 to every import?

File: backend/routers/medical_documentation.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if medical_documentation is None:
+        medical_documentation = models.MedicalDocumentation(patient_id=user_id)
+        db.add(medical_documentation)
+        db.commit()
+        print("DOne")
+        
+    return medical_documentation
+    
+@router.get("/medical_documentation")
+async def get_medical_documentation(user_id: int = Header(1), db: Session = Depends(get_db)):
```
Comment (kacienk) [inline backend/routers/medical_documentation.py]: What about the case when doctor would like to see his patient medical documentation? In my opinion it should be a query parameter

File: backend/routers/medical_documentation.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    if db.query(models.MedicalDocumentation).filter(models.MedicalDocumentation.patient_id == user_id).first() is None:
```
Comment (kacienk) [inline backend/routers/medical_documentation.py]: If the query is empty we get internal server error, so possibly this expression raises an exception when no item found

File: backend/routers/medical_documentation.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        raise HTTPException(status_code=404, detail="User not found")
+    
+    return db.query(models.MedicalDocumentation).filter(models.MedicalDocumentation.patient_id == user_id).options(joinedload(models.MedicalDocumentation.medical_documentation_entries)).first()
```
Comment (kacienk) [inline backend/routers/medical_documentation.py]: I think you can reuse the query by saving it in a variable and then query for the entries

File: backend/routers/referral.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    
+@router.get("/referals/{referal_id}")
+async def get_prescription(referal_id: int, db: Session = Depends(get_db)):
+    referal = db.query(model.Referral).filter(model.Referral.id == referal_id).first()
+    if referal is None:
+        raise HTTPException(status_code=404, detail="Referal not found")
+    return referal
+
+@router.post("/referal")
+async def add_prescription(referal: schema.Referral, patient_id = Header(1), doctor_id = Header(1), db: Session = Depends(get_db)):
```
Comment (kacienk) [inline backend/routers/referral.py]: Data should be read from request body not header

**Review by kacienk — COMMENTED** [review]
The endpoint comments apply to every endpoint. Please add proper type hints

File: backend/models/__init__.py (hunk 1 above)
Comment (KacperST) [reply backend/models/__init__.py:6]: not every, but every in this file. It ensures that models are initialized before engine creates the database. Noqa F401 are just for removing flask warning

File: backend/routers/medical_documentation.py (hunk 2 above)
Comment (KacperST) [reply backend/routers/medical_documentation.py]: Yea, you're right

File: backend/routers/medical_documentation.py (hunk 4 above)
Comment (KacperST) [reply backend/routers/medical_documentation.py]: Good point

File: backend/routers/prescription.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+@router.get("/prescriptions")
+async def get_prescription(patient_id = Header(1), db: Session = Depends(get_db)):
```
Comment (dubielel) [inline backend/routers/prescription.py]: Move patient_id to Query params

File: backend/routers/referral.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        yield db
+    finally:
+        db.close()
+
+@router.get("/referrals")
+async def get_prescription(patient_id = Header(1), db: Session = Depends(get_db)):
+    response = db.query(model.Referral).filter(model.Referral.patient_id == patient_id).all()
+    return response
+    
+@router.get("/referals/{referal_id}")
```
Comment (dubielel) [inline backend/routers/referral.py]: correct url to be "/referrals/{referral_id}"

File: backend/routers/referral.py (hunk 5 above)
Comment (dubielel) [reply backend/routers/referral.py]: bump

File: backend/routers/referral.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+async def get_prescription(referal_id: int, db: Session = Depends(get_db)):
+    referal = db.query(model.Referral).filter(model.Referral.id == referal_id).first()
+    if referal is None:
+        raise HTTPException(status_code=404, detail="Referal not found")
+    return referal
+
+@router.post("/referal")
```
Comment (dubielel) [inline backend/routers/referral.py]: correct url to "/referral"

File: backend/routers/referral.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+async def add_prescription(referal: schema.Referral, patient_id = Header(1), doctor_id = Header(1), db: Session = Depends(get_db)):
+    entry = model.Referral(**referal.model_dump(),patient_id = patient_id, doctor_id = doctor_id)
+    db.add(entry)
+    db.commit()
+    return entry
+
+
+@router.delete("/referal/{referal_id}")
```
Comment (dubielel) [inline backend/routers/referral.py]: correct url to "/referral/{referral_id}"

Comment (Glonk223) [reply backend/routers/referral.py]: Yup

File: backend/routers/referral.py (hunk 8 above)
Comment (Glonk223) [reply backend/routers/referral.py]: Yup

File: backend/routers/referral.py (hunk 7 above)
Comment (Glonk223) [reply backend/routers/referral.py]: Yup

File: backend/routers/prescription.py (hunk 6 above)
Comment (Glonk223) [reply backend/routers/prescription.py]: Same thing for get referral?

File: backend/routers/referral.py (hunk 5 above)
Comment (Glonk223) [reply backend/routers/referral.py]: You're right

File: backend/routers/medical_documentation.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+async def get_medical_documentation(user_id: int = Query(None), db: Session = Depends(get_db)):
+    if user_id is None:
+        raise HTTPException(status_code=400, detail="ID not provided")
+    medical_documentation = db.query(models.MedicalDocumentation).filter(models.MedicalDocumentation.patient_id == user_id).first()
+    if medical_documentation is None:
+        raise HTTPException(status_code=404, detail="User not found")
+    return medical_documentation
+    
+
+@router.get("/medical_documentation/{medical_documentation_entry_id}")
```
Comment (kacienk) [inline backend/routers/medical_documentation.py:36]: I think we should standardize medical_documentations endpoints urls.
1.  Use plural form, change '_' to '-'. [Source](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5)
2. Medical documentations have their unique ID. Make MedicalDocumentations primary key also a foreign key pointing to user or use the unique ID. Either way, the {parameter} in this route should be MedicalDocumentation ID, not a MedicalDocumentationEntry ID. Same applies to every other medical documentation endpoint.
3. We cannot neglect doctors' responsibilities. User should only see his own documentation, doctors should be able to see all of them. That is why we should make endpoints at least for: listing documentation `medical-documentations/`, getting one documentation entries `medical-documentations/{medical-documentations-id}/entries`.

**Review by kacienk — APPROVED** [review]
Lgtm
